src/machine.ts

```diff
@@ -0,0 +1,122 @@
+import {
+  Machine,
+  assign,
+  send,
+} from 'xstate';
+import { syncToLatestBlock } from './utils';
+import {
+  SyncSchema,
+  SyncContext,
+} from './types';
+
+export const syncHandler = (_context, _event) => (callback, onReceive) => {
+  console.log('Sync handler instantiated');
+  const iterator = syncToLatestBlock();
+  const asyncCall: () => void = async () => {
+    for (;;) {
+      const block = await iterator.next();
+
+      const { value, done } = block;
+      if (done) {
+        console.log('Done!', value)
+        break;
+      }
+
+      if (value && !value.success) {
+        console.log('Erroed!', value.message);
+        callback('ERROR');
+        return;
+      }
+
+      if (value.type === 'reorg') {
+        console.log(value.message);
+        callback('REORG');
+        return;
+      }
+
+      if (value.type === 'finished') {
+        console.log('FINISHED!');
+        callback('DONE');
+      }
+
+      console.log(value);
```

Comment:
I feel we will log the same message multiple times. Should we put it in an else block?

Reply:
Done in d13009045fbf876a0edbb2c7a9cc24abe6132a7d